janitor/functions.py

```diff
@@ -1021,3 +1025,171 @@ def _round_to_fraction(number, denominator, digits=np.inf):
     df[colname] = df[colname].apply(_round_to_fraction_partial)
 
     return df
+
+
+@lru_cache(maxsize=32)
+def _convert_currency(
+    from_currency: str = None,
+    to_currency: str = None,
+    historical_date: date = None,
+):
+    """
+    Currency conversion for Pandas DataFrame column.
+    Helper function for `convert_currency` method.
+    The API used is: https://exchangeratesapi.io/
+    """
+
+    url = "https://api.exchangeratesapi.io"
+
+    if historical_date:
+        assert isinstance(historical_date, date) | isinstance(
```

Comment:
I think I changed the type-checking assert statements to use the `check` function instead. It's cleaner, and raises a TypeErrror rather than an AssertionError, which is clearer, and in case someone runs Python while disabling asserts (it's possible), the type-checking errors will be raised.

Reply:
Makes sense.. Changing asserts to use `check` instead now.